packages/flutter/lib/src/cupertino/modal_bottom_sheet.dart

```diff
@@ -0,0 +1,389 @@
+            transitionBackgroundColor ?? CupertinoColors.black),
+  );
+}
+
+class CupertinoModalBottomSheetRoute<T> extends ModalBottomSheetRoute<T> {
+  final Radius topRadius;
+  final Curve? previousRouteAnimationCurve;
+
+  // Background color behind all routes
+  // Black by default
+  final Color? transitionBackgroundColor;
+
+  CupertinoModalBottomSheetRoute({
+    required WidgetBuilder builder,
+    required WidgetWithChildBuilder containerBuilder,
+    double? closeProgressThreshold,
+    String? barrierLabel,
+    AnimationController? secondAnimationController,
+    Curve? animationCurve,
+    Color? modalBarrierColor,
+    bool bounce = true,
+    bool isDismissible = true,
+    bool enableDrag = true,
+    required bool expanded,
+    Duration? duration,
+    RouteSettings? settings,
+    ScrollController? scrollController,
+    this.transitionBackgroundColor,
+    this.topRadius = _default_top_radius,
+    this.previousRouteAnimationCurve,
+  })  : assert(expanded != null),
+        assert(isDismissible != null),
+        assert(enableDrag != null),
+        super(
+          closeProgressThreshold: closeProgressThreshold,
+          scrollController: scrollController,
+          containerBuilder: containerBuilder,
+          builder: builder,
+          bounce: bounce,
+          barrierLabel: barrierLabel,
+          secondAnimationController: secondAnimationController,
+          modalBarrierColor: modalBarrierColor,
+          isDismissible: isDismissible,
+          enableDrag: enableDrag,
+          expanded: expanded,
+          settings: settings,
+          animationCurve: animationCurve,
+          duration: duration,
+        );
+
+  @override
+  Widget buildTransitions(
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    final paddingTop = MediaQuery.of(context)?.padding.top ?? 0;
+    final distanceWithScale =
+        (paddingTop + _behind_widget_visible_height) * 0.9;
+    final offsetY = secondaryAnimation.value * (paddingTop - distanceWithScale);
+    final scale = 1 - secondaryAnimation.value / 10;
+    return AnimatedBuilder(
+      builder: (context, child) => Transform.translate(
+        offset: Offset(0, offsetY),
+        child: Transform.scale(
+          scale: scale,
+          child: child,
+          alignment: Alignment.topCenter,
+        ),
+      ),
+      child: child,
+      animation: secondaryAnimation,
+    );
+  }
+
+  @override
+  Widget getPreviousRouteTransition(BuildContext context,
+      Animation<double> secondaryAnimation, Widget child) {
+    return _CupertinoModalTransition(
+      secondaryAnimation: secondaryAnimation,
+      body: child,
+      animationCurve: previousRouteAnimationCurve,
+      topRadius: topRadius,
+      backgroundColor: transitionBackgroundColor ?? CupertinoColors.black,
+    );
+  }
+}
+
+class _CupertinoModalTransition extends StatelessWidget {
```

Comment:
Looking at the lines above, is this meant for the z-level retracting perspective move? If so, it could use a more descriptive name.

Reply:
Yes, It is the transition from the previous route that scales down. I am not sure how to call the transition.